src/v/cluster_link/replication/partition_data_queue.cc

```diff
@@ -15,8 +15,23 @@
 namespace cluster_link::replication {
 
 partition_data_queue::partition_data_queue(size_t max_buffered_bytes)
-  : _sem(max_buffered_bytes, "partition_data_queue") {}
+  : _current_max_buffered(max_buffered_bytes)
+  , _sem(max_buffered_bytes, "partition_data_queue") {}
 
+void partition_data_queue::update_max_buffered(size_t new_value) {
+    _gate.check();
+
+    if (new_value == _current_max_buffered) {
+        return;
+    }
+    if (new_value > _current_max_buffered) {
+        _sem.signal(new_value - _current_max_buffered);
+
+    } else {
+        _sem.consume(_current_max_buffered - new_value);
+    }
+    _current_max_buffered = new_value;
```

Comment:
nit: I think "current_max_buffered" gives an impression that that much is currently buffered, just call it "_max_buffered_bytes" or something?